addons/account/models/account_move_line.py

```diff
@@ -1847,94 +1840,483 @@ def get_accounting_rate(vals):
+                        amounts_list.append({'amount_residual_currency': aml.amount_residual_currency})
+                    exchange_max_date = max(exchange_max_date, aml.date)
+                exchange_diff_values = exchange_lines_to_fix._prepare_exchange_difference_move_vals(
+                    amounts_list,
+                    company=involved_amls.company_id,
+                    exchange_date=exchange_max_date,
+                )
+
+                # Exchange difference for cash basis entries.
+                caba_lines_to_reconcile = None
+                if is_cash_basis_needed(involved_amls.account_id):
+                    caba_lines_to_reconcile = involved_amls._add_exchange_difference_cash_basis_vals(exchange_diff_values)
+
+                # Prepare the exchange difference.
+                if exchange_diff_values['move_values']['line_ids']:
+                    exchange_diff_full_batch_index.append(fulL_batch_index)
+                    exchange_diff_values_list.append(exchange_diff_values)
+                    full_batch['caba_lines_to_reconcile'] = caba_lines_to_reconcile
+
+        # ==== Create the full exchange journal entries ====
+        exchange_moves = self._create_exchange_difference_moves(exchange_diff_values_list)
+        for fulL_batch_index, exchange_move in zip(exchange_diff_full_batch_index, exchange_moves):
+            full_batch = full_batches[full_batch_index]
+            amls = full_batch['amls']
+            full_batch['exchange_move'] = exchange_move
+            exchange_move_lines = exchange_move.line_ids.filtered(lambda line: line.account_id == amls.account_id)
+            full_batch['amls'] |= exchange_move_lines
+
+        # ==== Create the full reconcile ====
+        # Note we are using Command.link and not Command.set because Command.set is triggering an unlink that is
+        # slowing down the assignation of the co-fields. Indeed, unlink is forcing a flush.
+        full_reconcile_values_list = []
+        full_reconcile_fulL_batch_index = []
+        for fulL_batch_index, full_batch in enumerate(full_batches):
+            amls = full_batch['amls']
+            involved_partials = amls.matched_debit_ids + amls.matched_credit_ids
+            if full_batch['is_fully_reconciled']:
+                full_reconcile_values_list.append({
+                    'exchange_move_id': full_batch.get('exchange_move') and full_batch['exchange_move'].id,
+                    'partial_reconcile_ids': [Command.link(partial.id) for partial in involved_partials],
+                    'reconciled_line_ids': [Command.link(aml.id) for aml in amls],
+                })
+                full_reconcile_fulL_batch_index.append(fulL_batch_index)
+
+        self.env['account.full.reconcile']\
+            .with_context(
+                skip_invoice_sync=True,
+                skip_invoice_line_sync=True,
+                skip_account_move_synchronization=True,
+                check_move_validity=False,
```

Comment:
Why do we skip checking the move validity? Is it slowing down the process that much?

Reply:
The only think we write is `full_reconcile_id`. We don't need all those extra computations making the code 10x slower